File: connectors/connector-basecamp/src/main/java/com/taskadapter/connector/basecamp/UserResolver.java
package com.taskadapter.connector.basecamp;

import com.taskadapter.model.GUser;

public interface UserResolver {
    /**
     * Resolves a user by a user. Converts user into a Basecamp user. May return
     * <code>null</code> if no user found. May receive null as input.
     * 
     * @param user
     *            user to resolve.
     * @return resolved user.
     */
    GUser resolveUser(GUser user);
}

File: connectors/connector-basecamp-common/src/main/scala/com/taskadapter/connector/basecamp/UserResolver.scala
package com.taskadapter.connector.basecamp

import com.taskadapter.model.GUser

trait UserResolver {
  /**
    * Find user by login.
    *
    * @return resolved user or <code>null</code> if no user found. May receive null as input.
    */
    def findUserByLogin(loginName: String): GUser

    def findUserByDisplayName(displayName: String): GUser
}